pynest/nest/tests/test_topology_gamma_kernel.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+#
+# test_topology_gamma_kernel.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Test of events
+"""
+
+import numpy as np
+import scipy.stats as sts
+import unittest
+import nest
+import nest.topology as tp
+
+
+@nest.check_stack
+class TopologyGammaKernelTestCase(unittest.TestCase):
+    """Tests of gamma kernel in topology"""
+
+    def test_connectivity_profile_match(self):
+        """Match connectivity profile with PDF"""
+
+        nest.ResetKernel()
+
+        extent = [100., 100.]
+        npop = 5000
+        positions = np.random.uniform(-extent[0] / 2, extent[1] / 2, [npop, 2])
+        ncon = 10000
+        kappa = 3.
+        theta = 4.
+        radius = extent[0] / 2 * np.sqrt(2)
+        dx = .5
+        x = np.arange(0, radius + dx, dx)
+
+        ly1 = tp.CreateLayer({
+            'positions': [[0., 0.]],
+            'elements': 'iaf_psc_alpha',
+            'extent': extent,
+            'edge_wrap': True
+        })
+
+        ly2 = tp.CreateLayer({
+            'positions': positions.tolist(),
+            'elements': 'iaf_psc_alpha',
+            'extent': extent,
+            'edge_wrap': True
+        })
+
+        tp.ConnectLayers(ly1, ly2, projections={
+            'connection_type': 'divergent',
+            'allow_oversized_mask': True,
+            'mask': {
+                'circular': {
+                    'radius': radius
+                }
+            },
+            'kernel': {
+                'gamma': {
+                    'kappa': kappa,
+                    'theta': theta
+                }
+            },
+            'number_of_connections': ncon,
+            'allow_multapses': True,
+            'allow_autapses': True,
+        })
+
+        gamma = sts.gamma.pdf(x, a=kappa + 1, scale=theta)
+        gamma_normed = gamma / float(np.sum(gamma)) / dx
+
+        center = tp.FindCenterElement(ly1)
+        tgts = tp.GetTargetNodes(center, ly2)[0]
+        d = tp.Distance(center, tgts)
+
+        h = np.histogram(d, bins=x)[0]
+        h_normed = h / float(np.sum(h)) / dx
+        difference = np.diff([gamma_normed[:-1], h_normed], axis=0)
+        error_score = np.abs(np.median(difference))
```

Comment:
Instead of this rather *ad hoc* test, why not use `scipy.stats.kstest()` as a proper statistical test?

Reply:
That is my concerning. I still do not understand, why PDF of the Gamma distribution function shows not matching results with the distribution of targets as a function of distance calculated in NEST. 

Using the same value in NEST and in Python the distributions scored a low match (Fig. 1-3, blue and orange traces). Then I increased kappa by 1 for PDF in python and compared with the distribution calculated from NEST. Here, these distributions have a better matching score (Fig 1-3, blue and green traces)

In my view, the PDFs in python were calculated correctly but I do not understand why NEST Topology gave different output.

### Figures

![gamma_kappa_1_theta_10](https://user-images.githubusercontent.com/620984/37700500-17ee27a2-2cec-11e8-8bb5-4bae6e1f2bdf.png)
Fig 1.

----------

![gamma_kappa_2_theta_5](https://user-images.githubusercontent.com/620984/37700657-ada047da-2cec-11e8-97cf-ed70674b1dbb.png)
Fig 2.

----------

![gamma_kappa_3_theta_4](https://user-images.githubusercontent.com/620984/37700675-b62d83c2-2cec-11e8-922e-432b7ed2a485.png)
Fig 3.

Reply:
@babsey The number of connections you will find in NEST at a given distance depends not only on the kernel value at that distance, but also on the number of potential targets at that distance; for the test, you thus need to convolve the gamma kernel with this density and then apply the KS-test. For details, see [Daniel Hjertholm, *Statistical tests for connection algorithms for structured neural networks* ](https://brage.bibsys.no/xmlui/handle/11250/189117), esp. Chapter 4.